lib/AFP/AbstractFilePath/Internal/Common.hs

```diff
@@ -0,0 +1,692 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+-- This template expects CPP definitions for:
+--     MODULE_NAME = Posix | Windows
+--     IS_WINDOWS  = False | True
+
+module AFP.AbstractFilePath.Internal.MODULE_NAME where
+
+-- doctest
+import AFP.AbstractFilePath.Internal.Types
+    ()
+
+#ifdef WINDOWS
+import qualified AFP.Data.ByteString.Short.Word16 as BS
```

Comment:
Here's the juice... on Posix the import is `import qualified AFP.Data.ByteString.Short as BS`, which makes the module work for both platforms via CPP.